gst/nnstreamer/tensor_repo.h

```diff
@@ -0,0 +1,113 @@
+/**
+ * NNStreamer Tensor Repo Header
+ * Copyright (C) 2018 Jijoong Moon <jijoong.moon@samsung.com>
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Library General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Library General Public License for more details.
+ *
+ */
+/**
+ * @file	tensor_repo.h
+ * @date	17 Nov 2018
+ * @brief	tensor repo header file for NNStreamer, the GStreamer plugin for neural networks
+ * @see		https://github.com/nnsuite/nnstreamer
+ * @author	Jijoong Moon <jijoong.moon@samsung.com>
+ * @bug		No known bugs except for NYI items
+ *
+ */
+#ifndef __GST_TENSOR_REPO_H__
+#define __GST_TENSOR_REPO_H__
+
+#include <glib.h>
+#include <stdint.h>
+#include <tensor_common.h>
+#include "tensor_typedef.h"
+#include <gst/gst.h>
+
+G_BEGIN_DECLS
+
+/**
+ * @brief GstTensorRepo internal data structure.
+ *
+ * GstTensorRepo has GSlist of GstTensorData.
+ *
+ */
+
+typedef struct
+{
+  GstTensorConfig *config;
+  GstBuffer *buffer;
+  GCond cond;
+  GMutex lock;
+} GstTensorData;
```

Comment:
Please rename GstTensorData to GstTensorRepoData.
Other developes working on other elements may be confused by the name

Reply:
This will be changed in PR #883